pkg/grub/rootfs.cfg

```diff
@@ -223,7 +223,6 @@ function set_arm64_baremetal {
    if [ "$smb_vendor" = "raspberrypi" ]; then
       smbios -t 1 -s 5 --set smb_product
       if [ "$smb_product" = "rpi" ]; then
-         set_global dom0_console "console=tty0 console=ttyS1,115200"
```

Comment:
@rvs yes, I want to use the default dom0_console parameter from https://github.com/lf-edge/eve/blob/master/pkg/grub/rootfs.cfg#L204.

But after the test, I found a another new problem. UART root console works fine now, but if I use command: `eve enter` command and contaner name i.e. `debug` or `xen-tools`, `pillar` - I get garbage on uart and i can't exit from container shell.
```
PM_RSTS: 0x00001000
RPi: BOOTLOADER release VERSION:c2f8c388 DATE: Apr 29 2021 TIME: 17:11:29 BOOTMODE: 0x00000006 part: 0 BUILD_TIMESTAMP=1619712685 0x801c18d9 0x00d03114 0x000c1901
PM_RSTS: 0x00001000
part 00000000 reset_info 00000000
uSD voltage 3.3V
Initialising SDRAM 'Micron' 32Gb x2 total-size: 64 Gbit 3200
PCI reset
PCI reset
VLI: HUB2: 0xfff00000 0x24e6 MCU: 0xfff20000 0x15218
XHCI-STOP
xHC ver: 256 HCS: 05000420 fc000031 00e70004 HCC: 002841eb
xHC ver: 256 HCS: 05000420 fc000031 00e70004 HCC: 002841eb
xHC ports 5 slots 32 intrs 4
Boot mode: SD (01) order f4
SD HOST: 250000000 CTL0: 0x00000000 BUS: 100000 Hz actual: 100000 HZ div: 2500 (1250) status: 0x1fff0000 delay: 1080
SD HOST: 250000000 CTL0: 0x00000f00 BUS: 100000 Hz actual: 100000 HZ div: 2500 (1250) status: 0x1fff0000 delay: 1080
OCR c0ff8000 [2]
CID: 001b534d4544325335307a92570a0154
CSD: 400e00325b590003baff7f800a400000
SD: bus-width: 4 spec: 2 SCR: 0x02058483 0x00000000
SD HOST: 250000000 CTL0: 0x00000f04 BUS: 50000000 Hz actual: 41666666 HZ div: 6 (3) status: 0x1fff0000 delay: 2
MBR: 0x00000800,   73728 type: 0x0c
MBR: 0x00000001,    2047 type: 0xee
MBR: 0x00000000,       0 type: 0x00
MBR: 0x00000000,       0 type: 0x00
Trying partition: 0
lba: 2048 oem: 'mkfs.fat' volume: ' EVE        '
rsc 32 fat-sectors 567 c-count 72562 c-size 1 r-dir 2 r-sec 0
PM_RSTS: 0x00001000
Trying partition: 0
lba: 2048 oem: 'mkfs.fat' volume: ' EVE        '
rsc 32 fat-sectors 567 c-count 72562 c-size 1 r-dir 2 r-sec 0
Read config.txt bytes     1162 hnd 0x00000874
Read start4.elf bytes  2241440 hnd 0x00000885
Read fixup4.dat bytes     5411 hnd 0x00000877
Firmware: 51215b4f6e3966401ecd99652a35cf1c25069113 Sep 30 2021 19:21:03
0x00d03114 0x00000000 0x00000fff
MEM GPU: 32 ARM: 992 TOTAL: 1024
Starting start4.elf @ 0xfec00200 partition 0
PCI reset
+


U-Boot 2021.10 (Nov 01 2021 - 08:13:25 +0000)

DRAM:  7.9 GiB
RPI 4 Model B (0xd03114)
MMC:   mmcnr@7e300000: 1, mmc@7e340000: 0
Loading Environment from FAT... Unable to read "uboot.env" from mmc0:1... In:    serial
Out:   vidconsole
Err:   vidconsole
Net:   eth0: ethernet@7d580000
PCIe BRCM: link up, 5.0 Gbps x1 (SSC)
starting USB...
Bus xhci_pci: Register 5000420 NbrPorts 5
Starting the controller
USB XHCI 1.00
scanning bus xhci_pci for devices... 3 USB Device(s) found
       scanning usb for storage devices... 0 Storage Device(s) found
Hit any key to stop autoboot:  0
switch to partitions #0, OK
mmc0 is current device
Scanning mmc 0:1...
libfdt fdt_check_header(): FDT_ERR_BADMAGIC
Card did not respond to voltage select! : -110
Scanning disk mmcnr@7e300000.blk...
Disk mmcnr@7e300000.blk not ready
Scanning disk mmc@7e340000.blk...
Bad magic number for SquashFS image.
Found 6 disks
** Unable to read file ubootefi.var **
Failed to load EFI variables
BootOrder not defined
EFI boot manager: Cannot load any image
Found EFI removable media binary efi/boot/bootaa64.efi
1078784 bytes read in 66 ms (15.6 MiB/s)
libfdt fdt_check_header(): FDT_ERR_BADMAGIC
Booting /efi\boot\bootaa64.efi
Welcome to GRUB!

/EndEntire
file path: /HardwareVendor(e61d73b9-a384-4acc-aeab-82e828f3628b)[0: ]
/UnknownMessaging(1a)/UnknownMessaging(1a)
/HD(2,13000,96000,d2e034b9705cfc43,2,2)/File(\EFI\BOOT)/File(BOOTAA64.EFI)
/EndEntire
Welcome to GRUB!



                             GNU GRUB  version 2.03

 ┌────────────────────────────────────────────────────────────────────────────┐
 │*Boot 0.0.0-pr2340-85d55cca-kvm-arm64                                       │
 │ Set Boot Options                                                           │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 │                                                                            │
 └────────────────────────────────────────────────────────────────────────────┘

      Use the ▲ and ▼ keys to select which entry is highlighted.
      Press enter to boot the selected OS, `e' to edit the commands
      before booting or `c' for a command-line.
   The highlighted entry will be executed automatically in 0s.
  Booting `Boot 0.0.0-pr2340-85d55cca-kvm-arm64'

/boot/xen.efi console=dtuart sync_console dom0_mem=750M,max:750M
dom0_max_vcpus=1 dom0_vcpus_pin
EFI stub: Booting Linux Kernel...
EFI stub: Using DTB from configuration table
EFI stub: Exiting boot services and installing virtual address map...
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd083]
[    0.000000] Linux version 5.10.7-default (root@f91e4d6fe0b9) (gcc (Alpine 10.2.1_pre1) 10.2.1 20201203, GNU ld (GNU Binutils) 2.35.1) #1 SMP Mon Nov 1 07:26:20 UTC 2021
[    0.000000] Machine model: Raspberry Pi 4 Model B Rev 1.4
[    0.000000] Warning: PCIe ACS overrides enabled; This may allow non-IOMMU protected peer-to-peer DMA
[    0.000000] efi: EFI v2.80 by Das U-Boot
[    0.000000] efi: RTPROP=0x3cb2b040 SMBIOS=0x3cb27000 RNG=0x1e44f040 MEMRESERVE=0x1e44e040
[    0.000000] efi: seeding entropy pool
[    0.000000] Reserved memory: created CMA memory pool at 0x000000002c000000, size 64 MiB
[    0.000000] OF: reserved mem: initialized node linux,cma, compatible id shared-dma-pool
[    0.000000] NUMA: No NUMA configuration found
[    0.000000] NUMA: Faking a node at [mem 0x0000000000000000-0x00000001ffffffff]
[    0.000000] NUMA: NODE_DATA [mem 0x1fefe8300-0x1fefebfff]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000]   DMA32    [mem 0x0000000040000000-0x00000000ffffffff]
[    0.000000]   Normal   [mem 0x0000000100000000-0x00000001ffffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000000000fff]
[    0.000000]   node   0: [mem 0x0000000000001000-0x000000001e44efff]
[    0.000000]   node   0: [mem 0x000000001e44f000-0x000000001e44ffff]
[    0.000000]   node   0: [mem 0x000000001e450000-0x000000003cb26fff]
[    0.000000]   node   0: [mem 0x000000003cb27000-0x000000003cb27fff]
[    0.000000]   node   0: [mem 0x000000003cb28000-0x000000003cb2afff]
[    0.000000]   node   0: [mem 0x000000003cb2b000-0x000000003cb2dfff]
[    0.000000]   node   0: [mem 0x000000003cb2e000-0x000000003cb2efff]
[    0.000000]   node   0: [mem 0x000000003cb2f000-0x000000003cb32fff]
[    0.000000]   node   0: [mem 0x000000003cb33000-0x000000003df4ffff]
[    0.000000]   node   0: [mem 0x000000003df50000-0x000000003df5ffff]
[    0.000000]   node   0: [mem 0x000000003df60000-0x000000003dffffff]
[    0.000000]   node   0: [mem 0x0000000040000000-0x00000000fbffffff]
[    0.000000]   node   0: [mem 0x0000000100000000-0x00000001ffffffff]
[    0.000000] Zeroed struct page in unavailable ranges: 25 pages
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x00000001ffffffff]
[    0.000000] percpu: Embedded 23 pages/cpu s56984 r8192 d29032 u94208
[    0.000000] Detected PIPT I-cache on CPU0
[    0.000000] CPU features: detected: EL2 vector hardening
[    0.000000] CPU features: kernel page table isolation forced ON by KASLR
[    0.000000] CPU features: detected: Kernel page table isolation (KPTI)
[    0.000000] CPU features: detected: Spectre-v2
[    0.000000] CPU features: detected: Spectre-v4
[    0.000000] CPU features: detected: ARM errata 1165522, 1319367, or 1530923
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 2039808
[    0.000000] Policy zone: Normal
[    0.000000] Kernel command line: BOOT_IMAGE=/boot/kernel console=tty0 console=ttyS0,115200 console=hvc0 root=PARTUUID=b934e0d2-5c70-43fc-8409-920c6b2d66f0 dom0_mem=750M,max:750M dom0_max_vcpus=1 dom0_vcpus_pin eve_mem=600M,max:600M eve_max_vcpus=1 ctrd_mem=350M,max:350M ctrd_max_vcpus=1 change=500 rootdelay=3 rfkill.default_state=0 pcie_acs_override=downstream,multifunction
[    0.000000] Dentry cache hash table entries: 1048576 (order: 11, 8388608 bytes, linear)
[    0.000000] Inode-cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: mapped [mem 0x0000000038b27000-0x000000003cb27000] (64MB)
[    0.000000] Memory: 4982136K/8290304K available (15232K kernel code, 2976K rwdata, 6780K rodata, 5056K init, 781K bss, 257524K reserved, 65536K cma-reserved)
[    0.000000] random: get_random_u64 called from __kmem_cache_create+0x30/0x560 with crng_init=0
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=480 to nr_cpu_ids=4.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] arch_timer: cp15 timer(s) running at 54.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0xc743ce346, max_idle_ns: 440795203123 ns
[    0.000002] sched_clock: 56 bits at 54MHz, resolution 18ns, wraps every 4398046511102ns
[    0.000194] Console: colour dummy device 80x25
[    0.000398] printk: console [tty0] enabled
[    0.000428] serial port 0 not yet initialized
[    0.000476] Calibrating delay loop (skipped), value calculated using timer frequency.. 108.00 BogoMIPS (lpj=540000)
[    0.000488] pid_max: default: 32768 minimum: 301
[    0.000539] LSM: Security Framework initializing
[    0.000686] Mount-cache hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.000795] Mountpoint-cache hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.001789] rcu: Hierarchical SRCU implementation.
[    0.003590] Remapping and enabling EFI services.
[    0.003877] smp: Bringing up secondary CPUs ...
[    0.004142] Detected PIPT I-cache on CPU1
[    0.004183] CPU1: Booted secondary processor 0x0000000001 [0x410fd083]
[    0.004461] Detected PIPT I-cache on CPU2
[    0.004480] CPU2: Booted secondary processor 0x0000000002 [0x410fd083]
[    0.004729] Detected PIPT I-cache on CPU3
[    0.004748] CPU3: Booted secondary processor 0x0000000003 [0x410fd083]
[    0.004787] smp: Brought up 1 node, 4 CPUs
[    0.004813] SMP: Total of 4 processors activated.
[    0.004820] CPU features: detected: 32-bit EL0 Support
[    0.004826] CPU features: detected: CRC32 instructions
[    0.004832] CPU features: detected: 32-bit EL1 Support
[    0.018536] CPU: All CPU(s) started at EL2
[    0.018603] alternatives: patching kernel code
[    0.113634] node 0 deferred pages initialised in 100ms
[    0.114920] devtmpfs: initialized
[    0.119200] Registered cp15_barrier emulation handler
[    0.119228] Registered setend emulation handler
[    0.119237] KASLR enabled
[    0.119509] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.119526] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.120934] pinctrl core: initialized pinctrl subsystem
[    0.121763] SMBIOS 3.0 present.
[    0.121780] DMI: Unknown Unknown Product/Unknown Product, BIOS 2021.10 10/01/2021
[    0.122099] NET: Registered protocol family 16
[    0.122861] DMA: preallocated 1024 KiB GFP_KERNEL pool for atomic allocations
[    0.122969] DMA: preallocated 1024 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.123190] DMA: preallocated 1024 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.123216] audit: initializing netlink subsys (disabled)
[    0.123348] audit: type=2000 audit(0.120:1): state=initialized audit_enabled=0 res=1
[    0.123964] thermal_sys: Registered thermal governor 'fair_share'
[    0.123968] thermal_sys: Registered thermal governor 'bang_bang'
[    0.123977] thermal_sys: Registered thermal governor 'step_wise'
[    0.123982] thermal_sys: Registered thermal governor 'user_space'
[    0.124152] cpuidle: using governor ladder
[    0.124224] cpuidle: using governor menu
[    0.124357] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.124411] ASID allocator initialised with 32768 entries
[    0.125059] Serial: AMBA PL011 UART driver
[    0.137789] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.137809] HugeTLB registered 32.0 MiB page size, pre-allocated 0 pages
[    0.137815] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.137821] HugeTLB registered 64.0 KiB page size, pre-allocated 0 pages
[    0.138883] cryptd: max_cpu_qlen set to 1000
[    0.140824] ACPI: Interpreter disabled.
[    0.141608] iommu: Default domain type: Translated
[    0.141779] vgaarb: loaded
[    0.142015] SCSI subsystem initialized
[    0.142193] usbcore: registered new interface driver usbfs
[    0.142217] usbcore: registered new interface driver hub
[    0.142235] usbcore: registered new device driver usb
[    0.142316] usb_phy_generic phy: supply vcc not found, using dummy regulator
[    0.142500] pps_core: LinuxPPS API ver. 1 registered
[    0.142507] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.142518] PTP clock support registered
[    0.142586] EDAC MC: Ver: 3.0.0
[    0.142833] Registered efivars operations
[    0.143566] NetLabel: Initializing
[    0.143576] NetLabel:  domain hash size = 128
[    0.143581] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.143611] NetLabel:  unlabeled traffic allowed by default
[    0.143915] clocksource: Switched to clocksource arch_sys_counter
[    0.144031] VFS: Disk quotas dquot_6.6.0
[    0.144068] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.144359] simple-framebuffer 3e402000.framebuffer: framebuffer at 0x3e402000, 0x7e9000 bytes, mapped to 0x(____ptrval____)
[    0.144370] simple-framebuffer 3e402000.framebuffer: format=a8r8g8b8, mode=1920x1080x32, linelength=7680
[    0.153037] Console: switching to colour frame buffer device 240x67
[    0.161493] simple-framebuffer 3e402000.framebuffer: fb0: simplefb registered!
[    0.161629] pnp: PnP ACPI: disabled
[    0.164915] NET: Registered protocol family 2
[    0.165303] tcp_listen_portaddr_hash hash table entries: 4096 (order: 4, 65536 bytes, linear)
[    0.165426] TCP established hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.165668] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes, linear)
[    0.166476] TCP: Hash tables configured (established 65536 bind 65536)
[    0.166736] UDP hash table entries: 4096 (order: 5, 131072 bytes, linear)
[    0.166847] UDP-Lite hash table entries: 4096 (order: 5, 131072 bytes, linear)
[    0.167160] NET: Registered protocol family 1
[    0.167215] PCI: CLS 0 bytes, default 64
[    0.167989] hw perfevents: enabled with armv8_cortex_a72 PMU driver, 7 counters available
[    0.168150] kvm [1]: IPA Size Limit: 44 bits
[    0.168806] kvm [1]: vgic interrupt IRQ9
[    0.168920] kvm [1]: Hyp mode initialized successfully
[    0.171129] Initialise system trusted keyrings
[    0.171280] workingset: timestamp_bits=40 max_order=21 bucket_order=0
[    0.173758] zbud: loaded
[    0.174508] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.174808] 9p: Installing v9fs 9p2000 file system support
[    0.201569] Key type asymmetric registered
[    0.201607] Asymmetric key parser 'x509' registered
[    0.201666] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 246)
[    0.201802] io scheduler mq-deadline registered
[    0.201838] io scheduler kyber registered
[    0.201932] io scheduler bfq registered
[    0.204629] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    0.205067] brcm-pcie fd500000.pcie: host bridge /scb/pcie@7d500000 ranges:
[    0.205124] brcm-pcie fd500000.pcie:   No bus range found for /scb/pcie@7d500000, using [bus 00-ff]
[    0.205196] brcm-pcie fd500000.pcie:      MEM 0x0600000000..0x063fffffff -> 0x00c0000000
[    0.205264] brcm-pcie fd500000.pcie:   IB MEM 0x0000000000..0x00bfffffff -> 0x0400000000
[    0.275972] brcm-pcie fd500000.pcie: link up, 5.0 GT/s PCIe x1 (SSC)
[    0.276136] brcm-pcie fd500000.pcie: PCI host bridge to bus 0000:00
[    0.276185] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.276227] pci_bus 0000:00: root bus resource [mem 0x600000000-0x63fffffff] (bus address [0xc0000000-0xffffffff])
[    0.276317] pci 0000:00:00.0: [14e4:2711] type 01 class 0x060400
[    0.276402] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.278006] pci 0000:01:00.0: [1106:3483] type 00 class 0x0c0330
[    0.278096] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x00000fff 64bit]
[    0.278252] pci 0000:01:00.0: PME# supported from D0 D3hot
[    0.305242] pci 0000:00:00.0: BAR 14: assigned [mem 0x600000000-0x6000fffff]
[    0.305297] pci 0000:01:00.0: BAR 0: assigned [mem 0x600000000-0x600000fff 64bit]
[    0.307344] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.309362] pci 0000:00:00.0:   bridge window [mem 0x600000000-0x6000fffff]
[    0.311500] pcieport 0000:00:00.0: enabling device (0000 -> 0002)
[    0.313605] pcieport 0000:00:00.0: PME: Signaling with IRQ 46
[    0.315842] pcieport 0000:00:00.0: AER: enabled with IRQ 46
[    0.318625] efifb: probing for efifb
[    0.320801] efifb: framebuffer at 0x3e402000, using 8100k, total 8100k
[    0.322834] efifb: mode is 1920x1080x32, linelength=7680, pages=1
[    0.324887] efifb: scrolling: redraw
[    0.326890] efifb: Truecolor: size=8:8:8:8, shift=24:16:8:0
[    0.328924] fb0: switching to EFI VGA from simple
[    0.331102] Console: switching to colour dummy device 80x25
[    0.339807] Console: switching to colour frame buffer device 240x67
[    0.348253] fb0: EFI VGA frame buffer device
[    0.354098] Serial: 8250/16550 driver, 32 ports, IRQ sharing enabled
[    0.357566] bcm2835-aux-uart fe215040.serial: there is not valid maps for state default
[    0.357808] printk: console [ttyS0] disabled
[    0.357890] fe215040.serial: ttyS0 at MMIO 0xfe215040 (irq = 22, base_baud = 62499999) is a 16550
[    1.583849] printk: console [ttyS0] enabled
[    1.588652] Serial: AMBA driver
[    1.592091] msm_serial: driver initialized
[    1.597163] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    1.608585] brd: module loaded
[    1.616891] loop: module loaded
[    1.636789] drbd: initialized. Version: 8.4.11 (api:1/proto:86-101)
[    1.643206] drbd: built-in
[    1.646013] drbd: registered as block device major 147
[    1.651867] bcm2835-power bcm2835-power: Broadcom BCM2835 power domains driver
[    1.660172] Rounding down aligned max_sectors from 4294967295 to 4294967288
[    1.667384] db_root: cannot open: /etc/target
[    1.672600] libphy: Fixed MDIO Bus: probed
[    1.677242] tun: Universal TUN/TAP device driver, 1.6
[    1.682780] bcmgenet fd580000.ethernet: GENET 5.0 EPHY: 0x0000
[    1.703922] libphy: bcmgenet MII bus: probed
[    1.783946] unimac-mdio unimac-mdio.-19: Broadcom UniMAC MDIO bus
[    1.791145] usbcore: registered new interface driver usb-storage
[    1.797468] mousedev: PS/2 mouse device common for all mice
[    1.803589] i2c /dev entries driver
[    1.807884] bcm2835-wdt bcm2835-wdt: Broadcom BCM2835 watchdog timer
[    1.815027] sdhci: Secure Digital Host Controller Interface driver
[    1.821334] sdhci: Copyright(c) Pierre Ossman
[    1.825842] Synopsys Designware Multimedia Card Interface Driver
[    1.832101] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.838537] ledtrig-cpu: registered to indicate activity on CPUs
[    1.845072] hid: raw HID events driver (C) Jiri Kosina
[    1.850368] usbcore: registered new interface driver usbhid
[    1.856056] usbhid: USB HID core driver
[    1.860243] bcm2835-mbox fe00b880.mailbox: mailbox enabled
[    1.867854] xt_time: kernel timezone is -0000
[    1.869481] mmc1: SDHCI controller on fe300000.mmcnr [fe300000.mmcnr] using PIO
[    1.872334] IPVS: Registered protocols ()
[    1.883913] IPVS: Connection hash table configured (size=4096, memory=64Kbytes)
[    1.891454] IPVS: ipvs loaded.
[    1.894655] ipip: IPv4 and MPLS over IPv4 tunneling driver
[    1.895110] mmc1: queuing unknown CIS tuple 0x80 (2 bytes)
[    1.900506] gre: GRE over IPv4 demultiplexor driver
[    1.909443] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.922517] ipt_CLUSTERIP: ClusterIP Version 0.8 loaded successfully
[    1.923945] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.938591] Initializing XFRM netlink socket
[    1.941451] mmc1: queuing unknown CIS tuple 0x80 (7 bytes)
[    1.945201] NET: Registered protocol family 10
[    1.954118] mmc1: queuing unknown CIS tuple 0x80 (3 bytes)
[    1.959634] Segment Routing with IPv6
[    1.972921] NET: Registered protocol family 17
[    1.979474] NET: Registered protocol family 15
[    1.986017] Bridge firewalling registered
[    1.992102] 8021q: 802.1Q VLAN Support v1.8
[    1.998404] 9pnet: Installing 9P2000 support
[    2.004747] Key type dns_resolver registered
[    2.011262] registered taskstats version 1
[    2.017411] Loading compiled-in X.509 certificates
[    2.022279] random: fast init done
[    2.026075] Loaded X.509 cert 'Build time autogenerated kernel key: a8ba26a341b69c69d931823cef93e0e19bca38ac'
[    2.031243] mmc1: new high speed SDIO card at address 0001
[    2.041812] Key type ._fscrypt registered
[    2.055437] Key type .fscrypt registered
[    2.061408] Key type fscrypt-provisioning registered
[    2.070784] uart-pl011 fe201000.serial: there is not valid maps for state default
[    2.080640] fe201000.serial: ttyAMA1 at MMIO 0xfe201000 (irq = 21, base_baud = 0) is a PL011 rev2
[    2.095048] raspberrypi-firmware soc:firmware: Attached to firmware from 2021-09-30T19:21:03
[    2.276529] xhci_hcd 0000:01:00.0: xHCI Host Controller
[    2.283937] xhci_hcd 0000:01:00.0: new USB bus registered, assigned bus number 1
[    2.293748] xhci_hcd 0000:01:00.0: hcc params 0x002841eb hci version 0x100 quirks 0x0000000000000890
[    2.305494] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.10
[    2.315952] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.325339] usb usb1: Product: xHCI Host Controller
[    2.332342] usb usb1: Manufacturer: Linux 5.10.7-default xhci-hcd
[    2.340586] usb usb1: SerialNumber: 0000:01:00.0
[    2.347591] hub 1-0:1.0: USB hub found
[    2.353495] hub 1-0:1.0: 1 port detected
[    2.359773] xhci_hcd 0000:01:00.0: xHCI Host Controller
[    2.367122] xhci_hcd 0000:01:00.0: new USB bus registered, assigned bus number 2
[    2.376688] xhci_hcd 0000:01:00.0: Host supports USB 3.0 SuperSpeed
[    2.385186] usb usb2: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 5.10
[    2.395623] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.404997] usb usb2: Product: xHCI Host Controller
[    2.411981] usb usb2: Manufacturer: Linux 5.10.7-default xhci-hcd
[    2.420216] usb usb2: SerialNumber: 0000:01:00.0
[    2.427140] hub 2-0:1.0: USB hub found
[    2.433003] hub 2-0:1.0: 4 ports detected
[    2.524924] Waiting 3 sec before mounting root device...
[    2.540504] mmc0: SDHCI controller on fe340000.mmc [fe340000.mmc] using ADMA
[    2.644733] mmc0: new ultra high speed DDR50 SDXC card at address 59b4
[    2.653856] mmcblk0: mmc0:59b4 ED2S5 119 GiB
[    2.667210]  mmcblk0: p1 p2 p3 p4 p9
[    2.723929] usb 1-1: new high-speed USB device number 2 using xhci_hcd
[    2.906467] usb 1-1: New USB device found, idVendor=2109, idProduct=3431, bcdDevice= 4.21
[    2.916812] usb 1-1: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    2.926087] usb 1-1: Product: USB2.0 Hub
[    2.933120] hub 1-1:1.0: USB hub found
[    2.939151] hub 1-1:1.0: 4 ports detected
[    3.263928] usb 1-1.3: new low-speed USB device number 3 using xhci_hcd
[    3.439462] usb 1-1.3: New USB device found, idVendor=04f2, idProduct=0402, bcdDevice= 1.65
[    3.449972] usb 1-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    3.459412] usb 1-1.3: Product: USB Keyboard
[    3.465750] usb 1-1.3: Manufacturer: Chicony
[    3.490543] input: Chicony USB Keyboard as /devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.3/1-1.3:1.0/0003:04F2:0402.0001/input/input0
[    3.574146] hid-generic 0003:04F2:0402.0001: input,hidraw0: USB HID v1.11 Keyboard [Chicony USB Keyboard] on usb-0000:01:00.0-1.3/input0
[    3.621103] input: Chicony USB Keyboard System Control as /devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.3/1-1.3:1.1/0003:04F2:0402.0002/input/input1
[    3.704033] input: Chicony USB Keyboard Consumer Control as /devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.3/1-1.3:1.1/0003:04F2:0402.0002/input/input2
[    3.723618] hid-generic 0003:04F2:0402.0002: input,hidraw1: USB HID v1.11 Device [Chicony USB Keyboard] on usb-0000:01:00.0-1.3/input1
[    5.598058] VFS: Mounted root (squashfs filesystem) readonly on device 179:2.
[    5.612622] devtmpfs: mounted
[    5.625676] Freeing unused kernel memory: 5056K
[    5.654101] Run /sbin/init as init process
[    5.896219] binfmt_misc: module verification failed: signature and/or required key missing - tainting kernel
[    6.967241] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    6.982038] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    7.034789] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    7.050187] usbcore: registered new interface driver brcmfmac
[    7.205021] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac43455-sdio for chip BCM4345/6
[    7.206227] brcmfmac: brcmf_c_process_clm_blob: no clm_blob available (err=-2), device may have limited channels available
[    7.206685] brcmfmac: brcmf_c_preinit_dcmds: Firmware: BCM4345/6 wl0: Mar  1 2015 07:29:38 version 7.45.18 (r538002) FWID 01-6a2c8ad4
................   ..............   ................
 ................   ............   ................
              ....    .........   ....
    ................   .......   ................
     ................   .....   ................
                    ...   .   ....
        ................     ................
          ...............   ................

              Edge Virtualization Engine

................   ..............   ................
 ................   ............   ................
              ....    .........   ....
    ................   .......   ................
     ................   .....   ................
                    ...   .   ....
        ................     ................
          ...............   ................

              Edge Virtualization Engine
linuxkit-dca632fd1365 login: root (automatic login)

EVE is Edge Virtualization Engine

Take a look around and don't forget to use eve(1).
[   71.746186] bcmgenet fd580000.ethernet: configuring instance for external RGMII (RX delay)
[   71.760332] bcmgenet fd580000.ethernet keth0: Link is Down
[   71.768408] eth0: port 1(keth0) entered blocking state
[   71.781894] eth0: port 1(keth0) entered disabled state
[   71.792791] device keth0 entered promiscuous mode                                                                                                                                                                                         [   75.914139] bcmgenet fd580000.ethernet keth0: Link is Up - 1Gbps/Full - flow control off
[   75.927219] IPv6: ADDRCONF(NETDEV_CHANGE): keth0: link becomes ready
[   75.938682] eth0: port 1(keth0) entered blocking state
[   75.948812] eth0: port 1(keth0) entered forwarding state
[   75.959875] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
linuxkit-dca632fd1365:~# eve
Welcome to EVE!
  commands: enter [qube (assumed pillar)] [command (assumed sh)]
            enter-user-app <qube>
            exec qube command
            list
            status
            start <qube> (requires a qube to be in a destroyed state)
            pause <qube>
            resume <qube>
            destroy <qube>
            persist list
            persist attach <disk>
            firewall drop
            verbose on|off
linuxkit-dca632fd1365:~# eve status
TASK         PID     STATUS
pillar       1237    RUNNING
watchdog     1343    RUNNING
wlan         1397    RUNNING
ntpd         1190    RUNNING
vtpm         1287    RUNNING
wwan         1450    RUNNING
xen-tools    1517    RUNNING
debug        1044    RUNNING
guacd        1096    RUNNING
newlogd      1143    RUNNING
linuxkit-dca632fd1365:~# ctr -n services.linuxkit tasks exec --tty --exec-id ttytry debug sh -l

▒▒▒=▒]▒E7ᶖ▒▒.▒ٵ▒5mm▒▒▒▒▒j▒▒▒▒▒Ph=5-Ӂ▒▒▒▒sW▒▒#▒▒▒▒]▒E7ᶖ▒▒.▒ٵ▒5mm▒▒▒▒▒j▒▒▒▒]Ph=5-
Ӂ▒▒▒▒sW▒▒#▒▒▒▒]▒E7ᶖ▒▒.▒ٵ▒5mm▒▒▒▒▒▒ٍ▒▒▒▒▒Ph=5-
Ӂ▒▒▒▒sW▒▒#▒▒▒▒]▒E7ᶖ▒▒.▒ٵ▒5mm▒▒▒▒▒▒▒▒▒▒▒Ph=5-
Ӂ▒▒▒▒sW▒▒#▒▒▒▒]▒E7ᶖ▒▒.▒ٵ▒5mm▒▒▒▒▒j▒▒▒▒▒Ph=5-
Ӂ▒▒▒▒sW▒▒#▒▒▒▒]▒E7ᶖ▒▒.▒ٵ▒55▒▒9▒▒7▒▒▒▒▒▒vo▒▒▒X▒wE▒▒:G▒▒Ӵ▒▒▒l
```

`ctr -n services.linuxkit tasks exec --tty --exec-id ttytry debug sh -l` is analog for command `eve enter debug`

Is this a problem on the Linuxkit side? Or something else?